be/src/storage/rowset/segment_v2/rle_page.h

```diff
@@ -140,7 +140,7 @@ class RlePageBuilder final : public PageBuilder {
     size_t _count;
     bool _finished;
     int _bit_width;
-    RleEncoder<CppType>* _rle_encoder;
+    std::unique_ptr<RleEncoder<CppType>> _rle_encoder;
```

Comment:
I don't think you need a `std::unique_ptr` here, it's exception-safe already.  What you need to do is ensure that `ColumnReader` will be freed.

Reply:
yes, but i think std::unique_ptr would be better

Reply:
Ok, it's just a kind reminder here. you can still replace it with `std::unique_ptr`, but I suggest submitting it in a different PR since it's more about coding style rather than exception-safe.